unified_planning/model/timing.py

```diff
@@ -212,6 +218,20 @@ def __init__(self, lower: FNode, upper: FNode, is_left_open: bool = False, is_ri
         Duration.__init__(self)
         Interval.__init__(self, lower, upper, is_left_open, is_right_open)
 
+    def __eq__(self, oth: object) -> bool:
+        if isinstance(oth, DurationInterval):
+            return self._lower == oth._lower and self._upper == oth._upper and self._is_left_open == oth._is_left_open and self._is_right_open == oth._is_right_open
+        else:
+            return False
+
+    def __hash__(self) -> int:
+        res = hash(self._lower) + hash(self._upper)
+        if self._is_left_open:
+            res ^= hash('is_left_open')
+        if self._is_right_open:
+            res ^= hash('is_right_open')
```

Comment:
I appreciate the thought of using XOR here, but there is a better way if you want good hashes (and we generally do not have good hashes in the UP ATM): https://stackoverflow.com/questions/29435556/how-to-combine-hash-codes-in-in-python3

```python
def __hash__(self) -> int:
    return hash((self._lower, self.upper, self._is_left_open, self._is_right_open))
```

However, this is a minor thing